axon-server-connector/src/main/java/org/axonframework/axonserver/connector/query/AxonServerQueryBus.java

```diff
@@ -128,34 +155,57 @@ public AxonServerQueryBus(AxonServerConnectionManager axonServerConnectionManage
+                                       completableFuture.complete(serializer.deserializeResponse(queryResponse));
                                    }
-
-                                   @Override
-                                   public void onError(Throwable throwable) {
-                                       logger.warn("Received error while waiting for first response: {}", throwable.getMessage(), throwable);
-                                       completableFuture.completeExceptionally(new RemoteQueryException(ErrorCode.QUERY_DISPATCH_ERROR.errorCode(), ErrorMessage.newBuilder().setMessage("No result from query executor").build()));
+                               }
+
+                               @Override
+                               public void onError(Throwable throwable) {
+                                   logger.warn("Received error while waiting for first response: {}",
+                                               throwable.getMessage(),
+                                               throwable);
+                                   completableFuture.completeExceptionally(new AxonServerQueryDispatchException(
+                                           ErrorCode.QUERY_DISPATCH_ERROR.errorCode(),
+                                           ExceptionSerializer.serialize(configuration.getClientId(), throwable)
+                                   ));
+                               }
+
+                               @Override
+                               public void onCompleted() {
+                                   if (!completableFuture.isDone()) {
+                                       completableFuture.completeExceptionally(new AxonServerQueryDispatchException(
+                                               ErrorCode.QUERY_DISPATCH_ERROR.errorCode(),
+                                               ErrorMessage.newBuilder()
+                                                           .setMessage("No result from query executor")
+                                                           .build()
+                                       ));
                                    }
-
-                                   @Override
-                                   public void onCompleted() {
-                                       if (!completableFuture.isDone()) {
-                                           completableFuture.completeExceptionally(new RemoteQueryException(ErrorCode.QUERY_DISPATCH_ERROR.errorCode(), ErrorMessage.newBuilder().setMessage("No result from query executor").build()));
-                                       }
-                                   }
-                               });
-
+                               }
+                           });
+        } catch (Exception e) {
+            logger.warn("There was a problem issuing a query {}.", interceptedQuery, e);
+            ErrorMessage errorMessage = ExceptionSerializer.serialize(configuration.getClientId(), e);
```

Comment:
I'm suggesting to use instead
ErrorCode.QUERY_DISPATCH_ERROR.convert(configuration.getClientId(), e) 
to simplify creation of exception

Reply:
@m1l4n54v1c suggested that exact line right now. And that's what we're doing in the `AxonServerCommandBus` too.